pytorch_lightning/trainer/training_loop.py

```diff
@@ -410,8 +410,9 @@ def train(self):
                 self.interrupted = True
                 self.on_keyboard_interrupt()
 
-                for proc in self.interactive_ddp_procs:
-                    subprocess.Popen.kill(proc)
+                if self.global_rank == 0:
+                    for proc in self.interactive_ddp_procs:
+                        subprocess.Popen.kill(proc)
```

Comment:
shouldn't it go into the run_training_teardown method?

Reply:
no because you still want certain things to happen with every process.

Reply:
oh wait, i see what you meant. i changed it. better now?

Reply:
not sure actually. I guess the "killing" should be the very last thing, after teardown has fully completed, right?
anyway not an expert here, so whatever works for you :)

Reply:
ah yes, i see now. That indeed should be the last thing since it will interrupt every other processes teardown. makes sense!